### ISSUE TITLE
vmagent 1.85.3 unable to start on armv5 kernel 2.6.34 - accept: function not implemented (golang issue)

### ISSUE THREAD

**Issue by jadbaz** [VictoriaMetrics/VictoriaMetrics#3570]
**Describe the bug**
When running it on armv5 with Linux kernel 2.6
```
# uname -a
Linux SDcard 2.6.34 #17 PREEMPT Fri Nov 11 15:06:02 MST 2011 armv5tel GNU/Linux
```

**To Reproduce**
Run the same command in the same environment
`./vmagent-prod --remoteWrite.url https://prom.com`

**Expected behavior**
Expected to run normally

**Logs**
```
SDcard:~/vmagent# ./vmagent-prod --remoteWrite.url https://prom.com
2022-12-29T12:10:43.057Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: vmagent-20221220-225637-tags-v1.85.3-0-gf443fad56
2022-12-29T12:10:43.058Z        info    VictoriaMetrics/lib/logger/flag.go:13   command-line flags
2022-12-29T12:10:43.059Z        info    VictoriaMetrics/lib/logger/flag.go:20     -remoteWrite.url="secret"
2022-12-29T12:10:43.060Z        info    VictoriaMetrics/app/vmagent/main.go:103 starting vmagent at ":8429"...
2022-12-29T12:10:43.062Z        info    VictoriaMetrics/lib/memory/memory.go:42 limiting caches to 77158809 bytes, leaving 51439207 bytes to the OS according to -memory.allowedPercent=60
2022-12-29T12:10:43.070Z        info    VictoriaMetrics/lib/persistentqueue/fastqueue.go:59     opened fast persistent queue at "vmagent-remotewrite-data/persistent-queue/1_E922287E695F1551" with maxInmemoryBlocks=77, it contains 0 pending bytes
2022-12-29T12:10:43.074Z        info    VictoriaMetrics/app/vmagent/remotewrite/client.go:169   initialized client for -remoteWrite.url="1:secret-url"
2022-12-29T12:10:43.076Z        info    VictoriaMetrics/app/vmagent/main.go:130 started vmagent in 0.016 seconds
2022-12-29T12:10:43.077Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:96 starting http server at http://127.0.0.1:8429/
2022-12-29T12:10:43.078Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:97 pprof handlers are exposed at http://127.0.0.1:8429/debug/pprof/
2022-12-29T12:10:43.085Z        panic   VictoriaMetrics/lib/httpserver/httpserver.go:147        FATAL: cannot serve http at :8429: accept tcp4 0.0.0.0:8429: accept: function not implemented
panic: FATAL: cannot serve http at :8429: accept tcp4 0.0.0.0:8429: accept: function not implemented

goroutine 25 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0x5a802a, 0x5}, {0xce6540, 0x5d}, 0x4)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:281 +0xbbc
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0x5a802a, 0x5}, {0x5bc515, 0x22}, {0xc5cf20, 0x2, 0x2})
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x478
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.serveWithListener({0x5a7f30, 0x5}, {0x717540, 0xd43c80}, 0x676048)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:147 +0x2ac
github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.Serve({0x5a7f30, 0x5}, 0x676048)
        github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:110 +0x4bc
created by main.main
        github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent/main.go:128 +0x54c
```

**Version**
The line returned when passing `--version` command line flag to the binary:
```
# ./vmagent-prod --version
vmagent-20221220-225637-tags-v1.85.3-0-gf443fad56
```

**Used command-line flags**
```./vmagent-prod --remoteWrite.url https://prom.com```

# Debugging
This is almost certainly the same issue as https://github.com/prometheus/node_exporter/issues/2476 relating to https://github.com/golang/go/issues/57333. 

## Fix
Wait for either of these issues to be closed and released
- 1.18 (released as part of 1.18.10): https://github.com/golang/go/issues/57338 - 
- 1.19 (released as part of 1.19.5): https://github.com/golang/go/issues/57339 -

**Comment by jadbaz** [comment]
So since this repo is keeping up with golang 1.19, I guess we should wait until 1.19.5. With the current release frequency, this should be expected early January

**Comment by jadbaz** [comment]
This is fixed with vmagent 1.86.1 which is compiled with golang 1.19.5

This is the same command above with no errors
```
# ./vmagent-linux-arm-v1.86.1 --remoteWrite.url https://prom.com
2023-01-13T08:25:12.160Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: vmagent-20230111-093824-tags-v1.86.1-0-g351fc152e
2023-01-13T08:25:12.161Z        info    VictoriaMetrics/lib/logger/flag.go:13   command-line flags
2023-01-13T08:25:12.162Z        info    VictoriaMetrics/lib/logger/flag.go:20     -remoteWrite.url="secret"
2023-01-13T08:25:12.164Z        info    VictoriaMetrics/app/vmagent/main.go:102 starting vmagent at ":8429"...
2023-01-13T08:25:12.167Z        info    VictoriaMetrics/lib/memory/memory.go:42 limiting caches to 77158809 bytes, leaving 51439207 bytes to the OS according to -memory.allowedPercent=60
2023-01-13T08:25:14.261Z        info    VictoriaMetrics/lib/persistentqueue/fastqueue.go:59     opened fast persistent queue at "vmagent-remotewrite-data/persistent-queue/1_E922287E695F1551" with maxInmemoryBlocks=77, it contains 0 pending bytes
2023-01-13T08:25:14.270Z        info    VictoriaMetrics/app/vmagent/remotewrite/client.go:169   initialized client for -remoteWrite.url="1:secret-url"
2023-01-13T08:25:14.274Z        info    VictoriaMetrics/app/vmagent/main.go:128 started vmagent in 2.109 seconds
2023-01-13T08:25:14.276Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:96 starting http server at http://127.0.0.1:8429/
2023-01-13T08:25:14.278Z        info    VictoriaMetrics/lib/httpserver/httpserver.go:97 pprof handlers are exposed at http://127.0.0.1:8429/debug/pprof/
```

Thanks for keeping up with these issues
